main.go

```diff
@@ -2,84 +2,51 @@ package main
 
 import (
 	"context"
+	"fmt"
+	"log/slog"
 	"net/http"
 	"os"
 	"os/signal"
 	"syscall"
 	"time"
 
 	"github.com/ministryofjustice/opg-go-common/env"
-	"github.com/ministryofjustice/opg-go-common/logging"
 	"github.com/ministryofjustice/opg-go-common/template"
 	"github.com/ministryofjustice/opg-sirius-lpa-frontend/internal/server"
 	"github.com/ministryofjustice/opg-sirius-lpa-frontend/internal/sirius"
+	"github.com/ministryofjustice/opg-sirius-lpa-frontend/internal/telemetry"
 	"github.com/ministryofjustice/opg-sirius-lpa-frontend/internal/templatefn"
-	"go.opentelemetry.io/contrib/detectors/aws/ecs"
 	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
-	"go.opentelemetry.io/contrib/propagators/aws/xray"
-	"go.opentelemetry.io/otel"
-	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
-	"go.opentelemetry.io/otel/sdk/trace"
 	"golang.org/x/mod/sumdb/dirhash"
-	"google.golang.org/grpc"
 )
 
-func initTracerProvider(ctx context.Context, logger *logging.Logger) func() {
-	resource, err := ecs.NewResourceDetector().Detect(ctx)
-	if err != nil {
-		logger.Fatal(err)
-	}
-
-	traceExporter, err := otlptracegrpc.New(ctx,
-		otlptracegrpc.WithInsecure(),
-		otlptracegrpc.WithEndpoint("0.0.0.0:4317"),
-		otlptracegrpc.WithDialOption(grpc.WithBlock()),
-	)
-	if err != nil {
-		logger.Fatal(err)
-	}
-
-	idg := xray.NewIDGenerator()
-	tp := trace.NewTracerProvider(
-		trace.WithResource(resource),
-		trace.WithSampler(trace.AlwaysSample()),
-		trace.WithBatcher(traceExporter),
-		trace.WithIDGenerator(idg),
-	)
-
-	otel.SetTracerProvider(tp)
-	otel.SetTextMapPropagator(xray.Propagator{})
-
-	return func() {
-		if err := tp.Shutdown(ctx); err != nil {
-			logger.Fatal(err)
-		}
-	}
+func fatal(err error, logger *slog.Logger) {
+	logger.Error(fmt.Sprintf("a fatal error occurred: %s", err.Error()))
+	os.Exit(1)
```

Comment:
You may find it nicer to structure with `main` calling something else like https://github.com/ministryofjustice/opg-modernising-lpa/blob/main/cmd/mlpa/main.go#L73, that way `defer`s will happen as expected